documentation/src/main/asciidoc/topics/marshalling.adoc

```diff
@@ -141,11 +178,69 @@ include::code_examples/CustomMarshaller.java[]
 include::config_examples/custom_marshaller.xml[]
 ----
 
-
 TIP: It's possible custom marshaller implementations to access a configured white list via the
 link:{javadocroot}/org/infinispan/commons/marshall/Marshaller.html#initialize(org.infinispan.commons.configuration.ClassWhiteList)
 method, which is called during startup.
 
+== Adding Java Classes to Deserialization White Lists
+{brandname} does not allow deserialization of custom payloads into Java classes
+for security reasons, which applies to JSON, XML, and marshalled `byte[]`
+content.
+
+You must add Java classes to a deserialization white list, either using system
+properties or specifying them in the {brandname} configuration.
+
+.System properties
+
+[source,options="nowrap",subs=attributes+]
+----
+// Specify a comma-separated list of fully qualified class names
+-Dinfinispan.deserialization.whitelist.classes=java.time.Instant,com.myclass.Entity
+
+// Specify a regular expression to match classes
+-Dinfinispan.deserialization.whitelist.regexps=.*
+----
+
+.Declarative
+
+[source,xml,options="nowrap",subs=attributes+]
+----
+include::config_examples/deserialization_whitelist.xml[]
+----
+
+== Storing Deserialized Objects
```

Comment:
We should probably change this section titile to something like:

"Storing Deserialized Objects in the Server"?